### ISSUE TITLE
treemacs-modify-theme should create the icons with the last treemacs-resize size

### ISSUE THREAD

**Issue by yyoncho** [Alexander-Miller/treemacs#712]
ATM if you do M-x lsp-treemacs-resize then the newly registered icons are not sized correctly.

**Comment by Alexander-Miller** [comment]
On it.

**Comment by Alexander-Miller** [comment]
The resize did for me in a quick test with one of lsp-treemacs' icons.: 
![resize](https://user-images.githubusercontent.com/5951157/91388097-f389e500-e836-11ea-87ec-90fc5f060bd8.png)

My guess is that the icons are not resized for you because of Emacs 27 new image manipulation capabilities in combination with some not quite right guard clause on my end.

What does check return for you:

```Elisp
(image-type-available-p 'imagemagick)
```

What do you see when you get the value of one of the icons you have created?

```Elisp
(treemacs-get-icon-value "enum")
```

**Comment by yyoncho** [comment]
> (image-type-available-p 'imagemagick)

This returns nil.

> (treemacs-get-icon-value 'java-test-class)


```
#("  " 0 1
  (display
   (image :type png :file "/home/yyoncho/.emacs.d/elpa/28.0/develop/lsp-java-20200804.1609/icons/vscode/class.png" :scale 1 :ascent center :background "unspecified-bg")
   img-selected
   (image :type png :file "/home/yyoncho/.emacs.d/elpa/28.0/develop/lsp-java-20200804.1609/icons/vscode/class.png" :scale 1 :ascent center :background "unspecified-bg")
   img-unselected
   (image :type png :file "/home/yyoncho/.emacs.d/elpa/28.0/develop/lsp-java-20200804.1609/icons/vscode/class.png" :scale 1 :ascent center :background "unspecified-bg")))
```

**Comment by Alexander-Miller** [comment]
How about this, does the following code insert the resized image as expect?

```Elisp
(let* ((img #("  " 0 1
              (display
               (image
                :type png
                :file "home/yyoncho/.emacs.d/elpa/28.0/develop/lsp-java-20200804.1609/icons/vscode/class.png"
                :scale 1
                :ascent center
                :background "unspecified-bg"))))
       (display (get-text-property 0 'display img))
       (plist (cdr display)))
  (plist-put plist :height 50)
  (plist-put plist :width 50)
  (insert img))
```

**Comment by yyoncho** [comment]
Yes. The resizing itself works. The issue is only if I register new icons they don't get the last size. If I do M-x treemacs-resize-icons after that they get fixed.

**Comment by Alexander-Miller** [comment]
Ah, I misunderstood. My bad.

That makes things a lot simpler, patch is coming in a bit.

**Comment by yyoncho** [comment]
Thank you. One question again for icons - why did you choose png over svg images?

**Comment by Alexander-Miller** [comment]
Let me put it this way: it's pretty bold of you to assume I knew enough about image types to make this a real choice. Lol

And a counter question: Would svg icons offer advantages other than better resizeability and could the current pngs be converted?

**Comment by spacekitteh** [comment]
> And a counter question: Would svg icons offer advantages other than better resizeability and could the current pngs be converted?

You can embed raster images into SVGs. SVG is a very flexible format :)

**Comment by Alexander-Miller** [comment]
@spacekitteh Please dumb it down a bit for me. Does that mean I'll be able to do something like programmatically add a symbol to an icon's corner to indicate uncommitted changes?

**Comment by yyoncho** [comment]
> Would svg icons offer advantages other than better resizeability 

I think this is the main advantage.

> and could the current pngs be converted?

I think that the majority of the icons in treemacs are originally svg. I am also not an expert in the field, but AFAIK if you convert png to svg they won't be with the same quality.
